### ISSUE TITLE
Users can rename sharing

### ISSUE THREAD

**Issue by bulletxt** [owncloud/core#15934]
userA shares folder A to userB giving him no read/write/delete permissions.   It seems though that userB can rename folder A.

After that, userA looses control over the folder .

**Comment by LukasReschke** [comment]
Thanks a lot for reporting issues back to us!

We need some more information to properly support you with your issue!

Can I ask you to follow our guidelines for submitting bugs as described here: https://github.com/owncloud/core/blob/master/CONTRIBUTING.md

**Comment by bulletxt** [comment]
Ubuntu 14.04 64-bit
Php 5.5.9
Apache/2.4.7 
mysql 5.5.41

config.php:
<?php
$CONFIG = array (
  'instanceid' => 'oc8idv3k3jjd',
  'passwordsalt' => 'aa',
  'secret' => 'aa',
  'trusted_domains' => 
  array (
    0 => 'localhost',
    1 => '192.168.30.44',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'http://localhost/owncloud',
  'dbtype' => 'mysql',
  'version' => '8.0.2.0',
  'dbname' => 'owncloud8',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_admin',
  'dbpassword' => 'aa',
  'installed' => true,
  'mail_smtpmode' => 'smtp',
  'mail_smtpsecure' => 'ssl',
  'mail_from_address' => 'backup',
  'mail_domain' => 'aa',
  'mail_smtpauthtype' => 'PLAIN',
  'mail_smtpauth' => 1,
  'mail_smtphost' => 'aa',
  'mail_smtpport' => '465',
  'mail_smtpname' => 'aa',
  'mail_smtppassword' => '000000',
);

**Comment by nickvergessen** [comment]
I can't reproduce that.

**Comment by LukasReschke** [comment]
Same here. Please provide us with complete reproduction steps as requested on https://github.com/owncloud/core/blob/master/CONTRIBUTING.md

**Comment by bulletxt** [comment]
After renaming, userA looses control over the folder because the menu "share" on that folder doesn't compare anymore

**Comment by LukasReschke** [comment]
Can you please fill out the issue template? – This is getting cumbersome, we really need detailed reproduction steps. See https://github.com/owncloud/core/issues/15933 for a good example where developers can see in a few seconds what the issue is and have an exact list of what the user did.

What I did here was:

1. Setup instance as user "admin"
2. Create a user "test" 
3. As user "admin" create a folder "shared"
4. Share folder "shared" with the user "test"
5. Login as user "test"
6. Rename the folder "shared" to "myfolder"
7. "admin" still sees "shared" and also sees the user in the sharing interface
8. "test" sees the folder "myfolder" which contains the same content as the folder "test

Without more detailed information we will not be able to debug this issue or help you as we can't reproduce your problem.

**Comment by bulletxt** [comment]
That's exactly what I'm doing but it fails on step 7. At step 7,  "admin" still sees "shared" folder BUT can't share the folder anymore to anyone and/or edit the sharing properties because the voice "share" won't appear on that folder anymore.  Only the link "Download" will appear.

I'm really not sure what other info can I provide .

**Comment by LukasReschke** [comment]
> I'm really not sure what other info can I provide .

https://raw.githubusercontent.com/owncloud/core/master/issue_template.md

- **Updated from an older ownCloud or fresh install:**
- **List of activated apps:**
- **Are you using external storage, if yes which one:** local/smb/sftp/...
- **Are you using encryption:** yes/no
- **Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
- Complete client configuration
- …

Also please provide us with the output of ` curl "http://localhost/master/ocs/v1.php/apps/files_sharing/api/v1/shares?format=json&shared_with_me=false" -u admin`, replace `admin` with the sharers' username and `localhost/master/` with where your ownCloud instance is located.

**Comment by bulletxt** [comment]
Its a new fresh install.
only "File" app is activated
no external storage
no encryption
no external user-backend

curl "http://localhost/owncloud/ocs/v1.php/apps/files_sharing/api/v1/shares?format=json&shared_with_me=false" -u impiegati
Enter host password for user 'impiegati':
{"ocs":{"meta":{"status":"ok","statuscode":100,"message":null},"data":[{"id":17,"item_type":"folder","item_source":"56","parent":null,"share_type":1,"share_with":"users","file_source":56,"file_target":"\/example","path":"\/ASD","permissions":1,"stime":1430319811,"expiration":null,"token":null,"storage":4,"mail_send":"0","uid_owner":"impiegati","share_with_displayname":"users","displayname_owner":"impiegati"}]}}

**Comment by LukasReschke** [comment]
The response seems to indicate that you did different steps than in https://github.com/owncloud/core/issues/15934#issuecomment-97457650, would it be possible for you to provide a complete list of steps that are required on a new (i.e. new setup) ownCloud instance to reproduce this problem? – I tried adjusting the permissions as well but locally still everything works fine.

**Comment by bulletxt** [comment]
0. create user "impiegati" part of group "xx". Create user "riccardo" part of group "usrers".
1. log in with user "impiegati"
2. create folder "ASD"
3. share folder "ASD" with group "users"
4. remove all permissions read/write/delete on this share 
5. log with user "riccardo"
6. rename shared folder "ASD" to "LOL"
7. log out user "riccardo"
8. log in with user "impiegati"
9. impiegati will still see folder "ASD"  BUT the passing the mouse over it will make only link "Download" appear. No link "share" will appear.

**Comment by LukasReschke** [comment]
Works here, does something appear in the browsers error console?

**Comment by LukasReschke** [comment]
@PVince81 Care to recheck as well on stable8?

**Comment by PVince81** [comment]
Hmm, it works for me on stable 8 (upcoming 8.0.3).
@bulletxt is the user "impiegati" also a member of "users" ? (this reminds me of https://github.com/owncloud/core/issues/15771 but there it's about an inned folder)

@bulletxt can you do a `select * from oc_filecache where path like '%ASD%'` and see what the "permissions" column contains ?

And then do a `select * from oc_share where item_source in (select fileid from oc_filecache where path like '%ASD%')` to get the matching share entry.

**Comment by bulletxt** [comment]
As said, "impiegati" is member of group "xx" that is different from group "users" .  Please remember im on 8.0.2 and not 8.0.3

for the select it returns 2 records:

"fileid","storage","path","path_hash","parent","name","mimetype","mimepart","size","mtime","storage_mtime","encrypted","unencrypted_size","etag","permissions"
"56","4","files_trashbin/files/ASD.d1430322358","22604d5665192d56392db0a4ca7faaa8","42","ASD.d1430322358","2","1","0","1430319806","1430319806","0","0","5540f2beef872","15"

"58","4","files/ASD","7c44e86e6241c7ec0c16fa9b4fee2b5f","33","ASD","2","1","0","1430322379","1430322379","0","0","5540fccb80e51","15"

**Comment by PVince81** [comment]
Hmm, still works for me in 8.0.2.
I also ran the oc_filecache query, and I can see that in my case the "permissions" are 31. In your case they are 15.

Did you change any sharing settings in the "Admin page" which might be different from the defaults ? Did you disable resharing ?
I tried to disable resharing globally in the admin page, but still same result: works for me.

In your report above you said "only the Files app is enabled". Does it mean you disabled the "Share files" app too ? (which is normally needed for sharing)

**Comment by bulletxt** [comment]
"appid","configkey","configvalue"
"activity","enabled","no"
"activity","installed_version","1.2.0"
"activity","ocsid","166038"
"activity","types","filesystem"
"backgroundjob","lastjob","3"
"core","global_cache_gc_lastrun","1430378874"
"core","installedat","1430313326.8285"
"core","lastcron","1430378882"
"core","lastupdateResult","{""version"":{},""versionstring"":{},""url"":{},""web"":{}}"
"core","lastupdatedat","1430378876"
"core","public_files","files_sharing/public.php"
"core","public_gallery","gallery/public.php"
"core","public_webdav","files_sharing/publicwebdav.php"
"core","remote_files","files/appinfo/remote.php"
"core","remote_webdav","files/appinfo/remote.php"
"core","shareapi_allow_links","yes"
"core","shareapi_allow_mail_notification","yes"
"core","shareapi_allow_public_notification","yes"
"core","shareapi_allow_resharing","no"
"core","shareapi_exclude_groups","yes"
"core","shareapi_exclude_groups_list","users"
"core","shareapi_only_share_with_group_members","no"
"files","enabled","yes"
"files","installed_version","1.1.9"
"files","types","filesystem"
"files_locking","enabled","yes"
"files_locking","installed_version",
"files_locking","types","filesystem"
"files_pdfviewer","enabled","yes"
"files_pdfviewer","installed_version","0.7"
"files_pdfviewer","ocsid","166049"
"files_pdfviewer","types",
"files_sharing","enabled","yes"
"files_sharing","installed_version","0.6.1"
"files_sharing","ocsid","166050"
"files_sharing","types","filesystem"
"files_texteditor","enabled","yes"
"files_texteditor","installed_version","0.4"
"files_texteditor","ocsid","166051"
"files_texteditor","types",
"files_trashbin","enabled","yes"
"files_trashbin","installed_version","0.6.2"
"files_trashbin","ocsid","166052"
"files_trashbin","types","filesystem"
"files_versions","enabled","yes"
"files_versions","installed_version","1.0.5"
"files_versions","ocsid","166053"
"files_versions","types","filesystem"
"files_videoviewer","enabled","yes"
"files_videoviewer","installed_version","0.1.3"
"files_videoviewer","ocsid","166054"
"files_videoviewer","types",
"firstrunwizard","enabled","yes"
"firstrunwizard","installed_version","1.1"
"firstrunwizard","ocsid","166055"
"firstrunwizard","types",
"gallery","enabled","no"
"gallery","installed_version","0.6.0"
"gallery","ocsid","166056"
"gallery","types",
"provisioning_api","enabled","yes"
"provisioning_api","installed_version","0.2"
"provisioning_api","types","filesystem"
"templateeditor","enabled","yes"
"templateeditor","installed_version","0.1"
"templateeditor","types",
"updater","enabled","yes"
"updater","installed_version","0.4"
"updater","ocsid","166059"
"updater","types",

**Comment by PVince81** [comment]
Okay, I was able to reproduce the problem on v8.0.2.

```
"core","shareapi_allow_resharing","no"
"core","shareapi_exclude_groups","yes"
"core","shareapi_exclude_groups_list","users"
```

Before sharing, need to go to he admin page in the "Sharing" section and do the following:
- disable resharing
- exclude the group "users" from sharing

Then continue the steps.
It seems something is messing up the permissions.

**Comment by PVince81** [comment]
Also happens on stable8 (upcoming v8.0.3)

Doesn't happen on master (upcoming 8.1), it seems to be fixed there.

**Comment by bulletxt** [comment]
Great, hopefully you can get this fixed for 8.0.3 or I'll just wait for 8.1

**Comment by nickvergessen** [comment]
8.0.3 is already in packaging, so it might be 8.0.4 or 8.1

**Comment by PVince81** [comment]
The next step would be to find out how it was fixed in 8.1 and whether it is easy to backport ("easy" as in "a fix that is not huge and doesn't risk breaking other stuff")

**Comment by rullzer** [comment]
Any news on finding the fix?

**Comment by PVince81** [comment]
Fix from 8.1 cannot be backported. Please upgrade to 8.1.1.
